sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Sink.scala

```diff
@@ -31,8 +31,11 @@ trait Sink {
    * this method is called more than once with the same batchId (which will happen in the case of
    * failures), then `data` should only be added once.
    *
-   * Note: You cannot apply any operators on `data` except consuming it (e.g., `collect/foreach`).
+   * Note 1: You cannot apply any operators on `data` except consuming it (e.g., `collect/foreach`).
    * Otherwise, you may get a wrong result.
+   *
+   * Note 2: The method is supposed to be executed synchronously, i.e. the method should only return
+   * after data is added to sink successfully.
```

Comment:
nit: data is added -> data is consumed. `add` may be not clear as you can also add `DataFrame` to a collection and do nothing.